packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,326 @@
+
+  /**
+   * The array size can be between 2 and 10,000.
+   * If you specify array properties for a job, it becomes an array job.
+   * For more information, see Array Jobs in the AWS Batch User Guide.
+   *
+   * @default - No array size
+   */
+  readonly arraySize?: number;
+
+  /**
+   * A list of container overrides in JSON format that specify the name of a container
+   * in the specified job definition and the overrides it should receive.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/APIReference/API_SubmitJob.html#Batch-SubmitJob-request-containerOverrides
+   *
+   * @default - No container overrides
+   */
+  readonly containerOverrides?: ContainerOverrides;
+
+  /**
+   * A list of dependencies for the job.
+   * A job can depend upon a maximum of 20 jobs.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/APIReference/API_SubmitJob.html#Batch-SubmitJob-request-dependsOn
+   *
+   * @default - No dependencies
+   */
+  readonly dependsOn?: JobDependency[];
+
+  /**
+   * The payload to be passed as parametrs to the batch job
+   *
+   * @default - No parameters are passed
+   */
+  readonly payload?: { [key: string]: any };
+
+  /**
+   * The number of times to move a job to the RUNNABLE status.
+   * You may specify between 1 and 10 attempts.
+   * If the value of attempts is greater than one,
+   * the job is retried on failure the same number of attempts as the value.
+   *
+   * @default - 1
+   */
+  readonly attempts?: number;
+
+  /**
+   * The timeout configuration for this SubmitJob operation.
+   * The minimum value for the timeout is 60 seconds.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/APIReference/API_SubmitJob.html#Batch-SubmitJob-request-timeout
+   *
+   * @default - No timeout
+   */
+  readonly timeout?: Duration;
+
+  /**
+   * The service integration pattern indicates different ways to call TerminateCluster.
+   *
+   * The valid value is either FIRE_AND_FORGET or SYNC.
+   *
+   * @default SYNC
+   */
+  readonly integrationPattern?: sfn.ServiceIntegrationPattern;
+}
+
+/**
+ * A Step Functions Task to run AWS Batch
+ */
+export class RunBatchJob implements sfn.IStepFunctionsTask {
+  private readonly integrationPattern: sfn.ServiceIntegrationPattern;
+
+  constructor(private readonly props: RunBatchJobProps) {
+    // validate integrationPattern
+    this.integrationPattern =
+      props.integrationPattern || sfn.ServiceIntegrationPattern.SYNC;
+
+    const supportedPatterns = [
+      sfn.ServiceIntegrationPattern.FIRE_AND_FORGET,
+      sfn.ServiceIntegrationPattern.SYNC
+    ];
+
+    if (!supportedPatterns.includes(this.integrationPattern)) {
+      throw new Error(
+        `Invalid Service Integration Pattern: ${this.integrationPattern} is not supported to call RunBatchJob.`
+      );
+    }
+
+    // validate arraySize limits
+    if (
+      props.arraySize !== undefined &&
+      (props.arraySize < 2 || props.arraySize > 10000)
+    ) {
```

Comment:
Single line is more readable -

```suggestion
    if (props.arraySize && (props.arraySize < 2 || props.arraySize > 10000)) {
```

Reply:
I was trying to do the same myself but the formatting is handled by Prettier

Reply:
Also, we have to compare to `undefined` since these are `number` type and we want to throw in case they are `zero`.

Reply:
Fair enough. Thanks!